drivers/gnss/gnss_seeed_air530z.c

```diff
@@ -0,0 +1,332 @@
+	};
+
+	data->uart_pipe = modem_backend_uart_init(&data->uart_backend, &uart_backend_config);
+}
+
+static int gnss_seeed_air530z_init_chat(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+
+	const struct modem_chat_config chat_config = {
+		.user_data = data,
+		.receive_buf = data->chat_receive_buf,
+		.receive_buf_size = sizeof(data->chat_receive_buf),
+		.delimiter = data->chat_delimiter,
+		.delimiter_size = ARRAY_SIZE(data->chat_delimiter),
+		.filter = NULL,
+		.filter_size = 0,
+		.argv = data->chat_argv,
+		.argv_size = ARRAY_SIZE(data->chat_argv),
+		.unsol_matches = unsol_matches,
+		.unsol_matches_size = ARRAY_SIZE(unsol_matches),
+	};
+
+	return modem_chat_init(&data->chat, &chat_config);
+}
+
+static void seeed_air530z_init_dynamic_script(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+
+	/* Air530z doesn't response to commands. Thus, response_matches_size = 0; */
+	data->dynamic_script_chat.request = data->dynamic_request_buf;
+	data->dynamic_script_chat.response_matches = NULL;
+	data->dynamic_script_chat.response_matches_size = 0;
+	data->dynamic_script_chat.timeout = 0;
+
+	data->dynamic_script.name = "PCAS";
+	data->dynamic_script.script_chats = &data->dynamic_script_chat;
+	data->dynamic_script.script_chats_size = 1;
+	data->dynamic_script.abort_matches = NULL;
+	data->dynamic_script.abort_matches_size = 0;
+	data->dynamic_script.callback = NULL;
+	data->dynamic_script.timeout = 5;
+}
+
+static int gnss_seeed_air530z_init(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+	const struct gnss_seeed_air530z_config *config = dev->config;
+	int ret;
+
+	LOG_DBG("initializing SEED Air530z");
+
+	ret = gnss_seeed_air530z_init_nmea0183_match(dev);
+	if (ret < 0) {
+		LOG_ERR("ERROR gnss_seeed_air530z_init_nmea0183_match");
+		return ret;
+	}
+
+	gnss_seeed_air530z_init_pipe(dev);
+
+	ret = gnss_seeed_air530z_init_chat(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	seeed_air530z_init_dynamic_script(dev);
+
+	ret = modem_pipe_open(data->uart_pipe);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = modem_chat_attach(&data->chat, data->uart_pipe);
+	if (ret < 0) {
+		modem_pipe_close(data->uart_pipe);
+		return ret;
+	}
+
+	ret = modem_chat_run_script(&data->chat, &init_script);
+	if (ret < 0) {
+		LOG_ERR("Failed to run init_script");
+		modem_pipe_close(data->uart_pipe);
+		return ret;
+	}
+
+	/* setup on-off gpio for power management */
+	if (!gpio_is_ready_dt(&config->on_off_gpio)) {
+		LOG_ERR("on-off GPIO device not ready");
+		return -ENODEV;
+	}
+
+	gpio_pin_configure_dt(&config->on_off_gpio, GPIO_OUTPUT_LOW);
+
+	return 0;
+}
+
+static int seeed_air530z_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+	const struct gnss_seeed_air530z_config *config = dev->config;
+	k_spinlock_key_t key;
+	int ret = -ENOTSUP;
+
+	key = k_spin_lock(&data->lock);
```

Comment:
do you need to lock here? it's not touching any data

Reply:
You are totally right, I expected to mange power by NMEA sentences and finally I noticed that this module only supports the on-off pin for powe management, I forgot to delete the lock